language/move-lang/src/typing/core.rs

```diff
@@ -817,6 +822,27 @@ pub fn make_function_type(
                 ])
             }
         }
+        FunctionVisibility::Friend(_) => {
+            if !in_current_module {
+                let is_in_friend_list = match &context.current_module {
+                    None => false,
+                    Some(current_mident) => {
+                        let minfo = context.module_info(m);
+                        minfo.friends.contains_key(current_mident)
+                    }
+                };
+                if !is_in_friend_list {
+                    let internal_msg = format!(
+                        "This function can only be called from a friend of module '{}'",
+                        m
+                    );
+                    context.error(vec![
+                        (loc, format!("Invalid call to '{}::{}'", m, f)),
+                        (defined_loc, internal_msg),
```

Comment:
For both this and script, I would actually point the location where the function was declared friend/script instead of the definition

Reply:
Fixed in 129744f, tests fixed in 72f08e3